src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -350,8 +350,6 @@ public void transferItemHold(UUID transactionId, UUID itemId) {
   public void finalCheckInItemHold(UUID transactionId, UUID servicePointId) {
     var transaction = fetchTransactionOfType(transactionId, ITEM);
 
-    verifyState(transaction, ITEM_RECEIVED, RECEIVE_UNANNOUNCED);
```

Comment:
this check is still needed. If loan is not yet checked-in then there will be a call `loanService.checkInItem(transaction, servicePointId);` and I guess it should occur only for transactions with the appropriate state